src/openfluid/ui/common/UIHelpers.hpp

```diff
@@ -188,28 +189,59 @@ inline QPixmap getImage(const QString& ImageName,const QString& ResourcePath,boo
 // =====================================================================
 
 
-inline QString createNewFile(QWidget* Parent, const QString& Path)
+inline QString createNewFile(QWidget* Parent, const QString& Path, bool WithLocationChoice=true)
 {
   if (Path.isEmpty())
   {
     return "";
   }
 
-  QString FileToCreate = QFileDialog::getSaveFileName(Parent,
-                                                      QApplication::translate("openfluid::ui::common","Create file"),
-                                                      Path,
-                                                      QApplication::translate("openfluid::ui::common","All files"));
-
+  QString FileToCreate = "";
+  if (WithLocationChoice)
+  {
+    QString FileName = QFileDialog::getSaveFileName(Parent,
+                                                QApplication::translate("openfluid::ui::common","Create file"),
+                                                Path,
+                                                QApplication::translate("openfluid::ui::common","All files"));
+    if (!FileName.isEmpty() && !FileName.isNull())
+    {
+      FileToCreate = FileName;
+    }
+  }
+  else
+  {
+    bool OK;
+    QString FileName = QInputDialog::getText(Parent, QApplication::translate("openfluid::ui::common","Add a file here"),
+                                             QApplication::translate("openfluid::ui::common","File name:"),
+                                             QLineEdit::Normal,
+                                             "",
+                                             &OK);
+
+    if (OK && !FileName.isEmpty())
+    {
+      FileToCreate = QString::fromStdString(openfluid::tools::FilesystemPath({Path.toStdString(),
+                                                                              FileName.toStdString()}).toNative());
+    }
+  }
   if (!FileToCreate.isEmpty())
   {
     auto NewFilePath = openfluid::tools::Path(FileToCreate.toStdString());
-    
+
     if (NewFilePath.isDirectory())
     {
       QMessageBox::critical(Parent,QApplication::translate("openfluid::ui::common","Error creating file"),
                            QApplication::translate("openfluid::ui::common","\"%1\" is an existing directory")
                              .arg(QString::fromStdString(NewFilePath.toNative())));
     }
+    // Checks if new path included in ref path
+    else if (!QString::fromStdString(NewFilePath.toGeneric()).contains(Path))
```

Comment:
If you try to check if Path is relative to NewFilePath, this may be the wrong way to do this as the contains method of QString does not guarantee that Path is at the end of NewFile Path. Maybe use openfluid::tools::FilesystemPath::contains() instead?